sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala

```diff
@@ -626,6 +624,120 @@ case class DescribeTableCommand(
+        AttributeReference("avg_col_len", StringType, nullable = true,
+          new MetadataBuilder().putString("comment",
+            "average length of the values of the column").build())(),
+        AttributeReference("max_col_len", StringType, nullable = true,
+          new MetadataBuilder().putString("comment",
+            "maximum length of the values of the column").build())(),
+        AttributeReference("comment", StringType, nullable = true,
+          new MetadataBuilder().putString("comment", "comment of the column").build())())
+    } else {
+      Seq(
+        AttributeReference("col_name", StringType, nullable = false,
+          new MetadataBuilder().putString("comment", "name of the column").build())(),
+        AttributeReference("data_type", StringType, nullable = false,
+          new MetadataBuilder().putString("comment", "data type of the column").build())(),
+        AttributeReference("comment", StringType, nullable = true,
+          new MetadataBuilder().putString("comment", "comment of the column").build())())
+    }
+  }
+
+  override def run(sparkSession: SparkSession): Seq[Row] = {
+    val catalog = sparkSession.sessionState.catalog
+    val resolver = sparkSession.sessionState.conf.resolver
+    val relation = sparkSession.table(table).queryExecution.analyzed
+    val field = {
+      relation.resolve(colNameParts, resolver).getOrElse {
+        throw new AnalysisException(s"Column ${UnresolvedAttribute(colNameParts).name} does not " +
+          s"exist")
+      }
+    }
+    if (!field.isInstanceOf[Attribute]) {
+      // If the field is not an attribute after `resolve`, then it's a nested field.
+      throw new AnalysisException(s"DESC TABLE COLUMN command is not supported for nested column:" +
+        s" ${UnresolvedAttribute(colNameParts).name}")
+    }
+
+    val catalogTable = catalog.getTempViewOrPermanentTableMetadata(table)
+    val colStats = catalogTable.stats.map(_.colStats).getOrElse(Map.empty)
+    val cs = colStats.get(field.name)
+
+    val comment = if (field.metadata.contains("comment")) {
+      Option(field.metadata.getString("comment"))
+    } else {
+      None
+    }
+
+    val fieldValues = if (isExtended) {
+      // Show column stats when extended or formatted is specified.
+      Seq(
+        field.name,
+        field.dataType.catalogString,
+        cs.flatMap(_.min.map(_.toString)).getOrElse("NULL"),
+        cs.flatMap(_.max.map(_.toString)).getOrElse("NULL"),
+        cs.map(_.nullCount.toString).getOrElse("NULL"),
+        cs.map(_.distinctCount.toString).getOrElse("NULL"),
+        cs.map(_.avgLen.toString).getOrElse("NULL"),
+        cs.map(_.maxLen.toString).getOrElse("NULL"),
+        comment.getOrElse("NULL"))
+    } else {
+      Seq(
+        field.name,
+        field.dataType.catalogString,
+        comment.getOrElse("NULL"))
+    }
+
+    Seq(Row(formatColumnInfo(fieldValues)))
```

Comment:
1. Column info is more than schema, it also has comment, column stats fields like min, max, etc.
2. Thanks for pointing that out, I'll fix it after we decide the output format.